### ISSUE TITLE
events.js:184 Uncaught TypeError: listener must be a function

### ISSUE THREAD

**Issue by lamosty** [Automattic/wp-calypso#6570]
I started getting `Uncaught TypeError: listener must be a function` error lately. First, I saw it on wp.com but after full reload it disappeared and I thought it's nothing big. However, today, I'm getting this error almost every time I boot up a new Calypso instance. Full reload usually helps.

@artpi noticed this error on WPcom in the `#js-errors` IRC channel and started debugging this so he might have more info.

#### Browser / OS version

Google Chrome Version 51.0.2704.63 (64-bit); Ubuntu 16.04

#### Screenshot / Video

I got a really long stack trace which I'm including below but it doesn't say much (doesn't include paths to files). Therefore, I'm also including a relevant screenshot:

![selection_031](https://cloud.githubusercontent.com/assets/4988512/16628864/c3bc16bc-43b3-11e6-8072-6855880189d7.png)

## Stack Trace 

```
EventEmitter.removeListener	@	events.js:184
componentWillUnmount	@	index.js:28
chainedFunction	@	ReactClass.js:533
(anonymous function)	@	makeAssimilatePrototype.js:15
unmountComponent	@	ReactCompositeComponent.js:395
unmountComponent	@	ReactReconciler.js:80
unmountComponent	@	ReactCompositeComponent.js:405
unmountComponent	@	ReactReconciler.js:80
unmountChildren	@	ReactChildReconciler.js:118
unmountChildren	@	ReactMultiChild.js:346
unmountComponent	@	ReactDOMComponent.js:925
unmountComponent	@	ReactReconciler.js:80
unmountChildren	@	ReactChildReconciler.js:118
unmountChildren	@	ReactMultiChild.js:346
unmountComponent	@	ReactDOMComponent.js:925
unmountComponent	@	ReactReconciler.js:80
unmountComponent	@	ReactCompositeComponent.js:405
unmountComponent	@	ReactReconciler.js:80
_updateRenderedComponent	@	ReactCompositeComponent.js:778
_performComponentUpdate	@	ReactCompositeComponent.js:753
updateComponent	@	ReactCompositeComponent.js:672
receiveComponent	@	ReactCompositeComponent.js:571
receiveComponent	@	ReactReconciler.js:127
_updateRenderedComponent	@	ReactCompositeComponent.js:775
_performComponentUpdate	@	ReactCompositeComponent.js:753
updateComponent	@	ReactCompositeComponent.js:672
receiveComponent	@	ReactCompositeComponent.js:571
receiveComponent	@	ReactReconciler.js:127
performUpdateIfNecessary	@	ReactCompositeComponent.js:583
performUpdateIfNecessary	@	ReactReconciler.js:160
runBatchedUpdates	@	ReactUpdates.js:151
perform	@	Transaction.js:136
perform	@	Transaction.js:136
perform	@	ReactUpdates.js:90
flushBatchedUpdates	@	ReactUpdates.js:177
closeAll	@	Transaction.js:202
perform	@	Transaction.js:149
batchedUpdates	@	ReactDefaultBatchingStrategy.js:63
enqueueUpdate	@	ReactUpdates.js:209
enqueueUpdate	@	ReactUpdateQueue.js:22
enqueueElementInternal	@	ReactUpdateQueue.js:206
(anonymous function)	@	ReactMount.js:244
scrollMonitor	@	ReactMount.js:232
_updateRootComponent	@	ReactMount.js:243
_renderSubtreeIntoContainer	@	ReactMount.js:338
render	@	ReactMount.js:386
renderWithReduxStore	@	index.js:25
plans	@	controller.jsx:101
(anonymous function)	@	index.js:472
nextEnter	@	index.js:304
navigation	@	controller.js:283
(anonymous function)	@	index.js:472
nextEnter	@	index.js:304
siteSelection	@	controller.js:226
(anonymous function)	@	index.js:472
nextEnter	@	index.js:304
retarget	@	ad-tracking.js:251
(anonymous function)	@	index.js:472
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
nextEnter	@	index.js:304
(anonymous function)	@	index.js:473
```

**Comment by yoavf** [comment]
Possibly duplicate of #6520

**Comment by yoavf** [comment]
Should be fixed with #6698

**Comment by lamosty** [comment]
> Should be fixed with #6698

Tested and yeah, fixed! Thanks.
